src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ZipFilesSummaryRepositoryTest.java

```diff
@@ -0,0 +1,124 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.entity;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.reports.ZipFileSummaryItem;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.reports.ZipFilesSummaryRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.helper.EnvelopeCreator;
+import uk.gov.hmcts.reform.bulkscanprocessor.helper.reports.zipfilesummary.Item;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.List;
+
+import static java.time.temporal.ChronoUnit.HOURS;
+import static java.time.temporal.ChronoUnit.MINUTES;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.bulkscanprocessor.entity.Status.COMPLETED;
+import static uk.gov.hmcts.reform.bulkscanprocessor.entity.Status.CONSUMED;
+import static uk.gov.hmcts.reform.bulkscanprocessor.entity.Status.PROCESSED;
+import static uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event.FILE_VALIDATION_FAILURE;
+import static uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event.ZIPFILE_PROCESSING_STARTED;
+
+@RunWith(SpringRunner.class)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@DataJpaTest
+public class ZipFilesSummaryRepositoryTest {
+
+    @Autowired
+    private ZipFilesSummaryRepository reportRepo;
+    @Autowired
+    private ProcessEventRepository eventRepo;
+    @Autowired
+    private EnvelopeRepository envelopeRepo;
+
+    @Test
+    public void should_join_envelopes_and_events_and_return_summary_by_date() {
+        // given
+        Instant createdDate = Instant.now();
```

Comment:
It may be worth using a fixed date to avoid surprises when the test runs around midnight.

Also, an extra case for such a scenario would be nice (file received on one day, but completed next day)

Reply:
@rkondratowicz updated dates in the existing test and added a new test for the scenario where a zipfile's received date and completed dates are different